pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -204,13 +237,61 @@ func TestHTTPScrapeClient_Scrape_ErrorCases(t *testing.T) {
 	}
 }
 
+func TestHTTPScrapeClient_ScrapeProto_ErrorCases(t *testing.T) {
+	testCases := []struct {
+		name            string
+		responseCode    int
+		responseErr     error
+		responseContext Stat
+		expectedErr     string
+	}{{
+		name:         "Non 200 return code",
+		responseCode: http.StatusForbidden,
+		expectedErr:  fmt.Sprintf(`GET request for URL %q returned HTTP status 403`, testURL),
+	}, {
+		name:         "Error got when sending request",
+		responseCode: http.StatusOK,
+		responseErr:  errors.New("upstream closed"),
+		expectedErr:  "upstream closed",
+	}}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			hClient := newTestHTTPClient(getProtoHTTPResponse(test.responseCode, test.responseContext, protoAcceptContent), test.responseErr)
+			sClient, err := newHTTPScrapeClient(hClient)
+			if err != nil {
+				t.Errorf("newHTTPScrapeClient=%v, want no error", err)
+			}
+			if _, err := sClient.Scrape(testURL); err != nil {
+				if !strings.Contains(err.Error(), test.expectedErr) {
+					t.Errorf("Got error message: %q, want to contain: %q", err.Error(), test.expectedErr)
+				}
+			} else {
+				t.Error("Expected error from newServiceScraperWithClient, got nil")
+			}
+		})
+	}
```

Comment:
```suggestion
			sClient, err := newHTTPScrapeClient(hClient)
			if err != nil {
				t.Fatalf("newHTTPScrapeClient=%v, want no error", err)
			}
			_, err := sClient.Scrape(testURL)
			if err == nil {
				t.Fatal("Got no error")
			}
			if !strings.Contains(err.Error(), test.expectedErr) {
				t.Errorf("Error = %q, want to contain: %q", err.Error(), test.expectedErr)
			}
		})
	}
```

Reply:
@vagababov it seems that this code part pre-existed, should we do these kind of refactorings in this PR?
For example there was a suggestion above to name the struct members in test cases data structures. If we look at 
[this file](https://github.com/knative/serving/blob/77c03f0f6f5f86b0cfe5023786351974403276ec/pkg/queue/breaker_test.go)  or t[his one](https://github.com/knative/serving/blob/master/cmd/autoscaler/main_test.go) (I am sure there are more), this convention is not followed. @markusthoemmes thoughts?

Reply:
Yeah, pull them into a separate PR if possible, it only distracts from this one.

Personally, I usually do a rough pass at the thing I'm trying to solve and if I notice any refactorings I need, I create PRs for those that precede the actual future PR.